### ISSUE TITLE
network error at the beginning

### ISSUE THREAD

**Issue by 2659494286** [szczyglis-dev/py-gpt#4]
File "/home/joker/anaconda3/lib/python3.11/site-packages/openai/_base_client.py", line 982, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/home/joker/anaconda3/lib/python3.11/site-packages/openai/_base_client.py", line 911, in _request
    return self._retry_request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/joker/anaconda3/lib/python3.11/site-packages/openai/_base_client.py", line 982, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/home/joker/anaconda3/lib/python3.11/site-packages/openai/_base_client.py", line 921, in _request
    raise APIConnectionError(request=request) from err

Error in GPT custom call: Connection error.

**Comment by 2659494286** [comment]
![image](https://github.com/szczyglis-dev/py-gpt/assets/15061456/f8cb363d-0131-444b-95d7-5afc9f090b19)
pip list:
Package                                  Version
---------------------------------------- ---------------
absl-py                                  2.0.0
aiobotocore                              2.9.0
aiofiles                                 22.1.0
aiohttp                                  3.9.1
aioitertools                             0.7.1
aiosignal                                1.3.1
aiosqlite                                0.18.0
alabaster                                0.7.12
altair                                   5.2.0
altgraph                                 0.17.3
anaconda-anon-usage                      0.4.2
anaconda-catalogs                        0.2.0
anaconda-client                          1.12.1
anaconda-cloud-auth                      0.1.3
anaconda-navigator                       2.5.0
anaconda-project                         0.11.1
aniso8601                                9.0.1
annotated-types                          0.6.0
anyio                                    3.7.1
appdirs                                  1.4.4
argon2-cffi                              21.3.0
argon2-cffi-bindings                     21.2.0
arrow                                    1.2.3
asgiref                                  3.7.2
astroid                                  2.14.2
astropy                                  5.1
asttokens                                2.0.5
astunparse                               1.6.3
async-timeout                            4.0.2
asyncio                                  3.4.3
atomicwrites                             1.4.0
attrs                                    22.2.0
autogen                                  1.0.16
Automat                                  20.2.0
autopep8                                 1.6.0
azure-cognitiveservices-speech           1.27.0
Babel                                    2.11.0
backcall                                 0.2.0
backoff                                  2.2.1
backports.functools-lru-cache            1.6.4
backports.tempfile                       1.0
backports.weakref                        1.0.post1
bayesian-optimization                    1.4.3
bcrypt                                   4.1.2
beautifulsoup4                           4.12.2
binaryornot                              0.4.4
black                                    0.0
bleach                                   6.0.0
bokeh                                    3.2.1
boltons                                  23.0.0
botocore                                 1.33.13
Bottleneck                               1.3.5
brotlipy                                 0.7.0
bs4                                      0.0.1
build                                    0.10.0
cachetools                               5.3.2
certifi                                  2022.12.7
cffi                                     1.15.1
cftime                                   1.6.3
chardet                                  4.0.0
charset-normalizer                       3.1.0
chroma-hnswlib                           0.7.2
chromadb                                 0.4.5
click                                    8.1.7
cloudpickle                              2.2.1
clyent                                   1.2.2
cmake                                    3.28.1
colorama                                 0.4.6
colorcet                                 3.0.1
coloredlogs                              15.0.1
comm                                     0.1.2
conda                                    23.7.4
conda-build                              3.26.1
conda-content-trust                      0.2.0
conda_index                              0.3.0
conda-libmamba-solver                    23.7.0
conda-pack                               0.6.0
conda-package-handling                   2.2.0
conda_package_streaming                  0.9.0
conda-repo-cli                           1.0.75
conda-token                              0.4.0
conda-verify                             3.4.2
constantly                               15.1.0
contourpy                                1.0.5
cookiecutter                             1.7.3
croniter                                 2.0.1
cryptography                             40.0.1
cssselect                                1.1.0
cycler                                   0.11.0
cytoolz                                  0.12.0
daal4py                                  2023.1.1
dashscope                                1.13.6
dask                                     2023.6.0
dataclasses-json                         0.6.3
datasets                                 2.16.1
datashader                               0.15.2
datashape                                0.5.4
ddgr                                     1.9
debugpy                                  1.6.7
decorator                                5.1.1
decord                                   0.6.0
defusedxml                               0.7.1
Deprecated                               1.2.14
diff-match-patch                         20200713
dill                                     0.3.6
diskcache                                5.6.3
distributed                              2023.6.0
distro                                   1.8.0
docker                                   7.0.0
docopt                                   0.6.2
docstring-to-markdown                    0.11
docutils                                 0.19
docx2txt                                 0.8
EbookLib                                 0.18
entrypoints                              0.4
et-xmlfile                               1.1.0
evdev                                    1.6.1
exceptiongroup                           1.2.0
executing                                0.8.3
faiss-cpu                                1.7.4
fastapi                                  0.99.1
fastjsonschema                           2.16.2
ffmpy                                    0.3.1
filelock                                 3.9.0
fire                                     0.5.0
flake8                                   6.0.0
FLAML                                    2.1.1
Flask                                    2.2.3
Flask-Cors                               3.0.10
Flask-RESTful                            0.3.9
flatbuffers                              23.5.26
fonttools                                4.25.0
frozenlist                               1.3.3
fsspec                                   2023.12.2
future                                   0.18.3
gast                                     0.5.4
gensim                                   4.3.0
glob2                                    0.7
gmpy2                                    2.1.2
google-auth                              2.25.2
google-auth-oauthlib                     1.1.0
google-pasta                             0.2.0
googleapis-common-protos                 1.62.0
gradio                                   3.50.2
gradio_client                            0.6.1
greenlet                                 3.0.1
grpcio                                   1.60.0
h11                                      0.14.0
h5py                                     3.9.0
HeapDict                                 1.0.1
holoviews                                1.17.1
html2text                                2020.1.16
httpcore                                 1.0.2
httptools                                0.6.1
httpx                                    0.25.2
huggingface-hub                          0.20.2
humanfriendly                            10.0
hvplot                                   0.8.4
hyperlink                                21.0.0
idna                                     3.4
imagecodecs                              2023.1.23
imageio                                  2.31.1
imagesize                                1.4.1
imbalanced-learn                         0.10.1
importlib-metadata                       6.3.0
importlib-resources                      6.1.1
incremental                              21.3.0
inflection                               0.5.1
iniconfig                                2.0.0
intake                                   0.6.8
intervaltree                             3.1.0
ipykernel                                6.25.0
ipython                                  8.15.0
ipython-genutils                         0.2.0
ipywidgets                               8.0.4
isort                                    5.9.3
itemadapter                              0.3.0
itemloaders                              1.0.4
itsdangerous                             2.0.1
jaraco.classes                           3.2.3
jedi                                     0.18.1
jeepney                                  0.8.0
jellyfish                                1.0.1
Jinja2                                   3.1.2
jinja2-time                              0.2.0
jmespath                                 0.10.0
joblib                                   1.3.2
json5                                    0.9.6
jsonpatch                                1.33
jsonpointer                              2.4
jsonschema                               4.17.3
jupyter                                  1.0.0
jupyter_client                           7.4.9
jupyter-console                          6.6.3
jupyter_core                             5.3.0
jupyter-events                           0.6.3
jupyter-server                           1.23.4
jupyter_server_fileid                    0.9.0
jupyter_server_ydoc                      0.8.0
jupyter-ydoc                             0.2.4
jupyterlab                               3.6.3
jupyterlab-pygments                      0.1.2
jupyterlab_server                        2.22.0
jupyterlab-widgets                       3.0.5
kaleido                                  0.2.1
keras                                    2.15.0
keyring                                  23.13.1
kiwisolver                               1.4.4
kubernetes                               29.0.0
langchain                                0.1.0
langchain-community                      0.0.11
langchain-core                           0.1.10
langchain-experimental                   0.0.49
langchain-openai                         0.0.2.post1
langsmith                                0.0.80
lazy_loader                              0.2
lazy-object-proxy                        1.6.0
libarchive-c                             2.9
libclang                                 16.0.6
libmambapy                               1.5.1
linkify-it-py                            2.0.0
lit                                      17.0.6
llama-hub                                0.0.69
llama-index                              0.9.29
llvmlite                                 0.40.0
lmdb                                     1.4.1
locket                                   1.0.0
lxml                                     5.1.0
lz4                                      4.3.2
Markdown                                 3.5.1
markdown-it-py                           2.2.0
MarkupSafe                               2.1.2
marshmallow                              3.20.1
matplotlib                               3.7.2
matplotlib-inline                        0.1.6
mccabe                                   0.7.0
mdit-py-plugins                          0.3.0
mdurl                                    0.1.2
mistune                                  0.8.4
mkl-fft                                  1.3.8
mkl-random                               1.2.4
mkl-service                              2.4.0
ml-dtypes                                0.2.0
mmh3                                     4.1.0
monotonic                                1.6
more-itertools                           9.1.0
mpmath                                   1.3.0
msgpack                                  1.0.3
multidict                                6.0.4
multipledispatch                         0.6.0
multiprocess                             0.70.14
munkres                                  1.1.4
mypy-extensions                          1.0.0
navigator-updater                        0.4.0
nbclassic                                0.5.5
nbclient                                 0.5.13
nbconvert                                6.5.4
nbformat                                 5.9.2
nest-asyncio                             1.5.8
netCDF4                                  1.6.5
networkx                                 3.2.1
nh3                                      0.2.15
nltk                                     3.8.1
notebook                                 6.5.4
notebook_shim                            0.2.2
numba                                    0.57.1
numexpr                                  2.8.4
numpy                                    1.26.2
numpydoc                                 1.5.0
oauthlib                                 3.2.2
onnxruntime                              1.16.3
openai                                   1.7.2
opencv-python                            4.8.1.78
opencv-python-headless                   4.8.1.78
openpyxl                                 3.1.2
opentelemetry-api                        1.22.0
opentelemetry-exporter-otlp-proto-common 1.22.0
opentelemetry-exporter-otlp-proto-grpc   1.22.0
opentelemetry-instrumentation            0.43b0
opentelemetry-instrumentation-asgi       0.43b0
opentelemetry-instrumentation-fastapi    0.43b0
opentelemetry-proto                      1.22.0
opentelemetry-sdk                        1.22.0
opentelemetry-semantic-conventions       0.43b0
opentelemetry-util-http                  0.43b0
opt-einsum                               3.3.0
orjson                                   3.9.10
overrides                                7.4.0
packaging                                23.2
pandas                                   2.1.4
pandocfilters                            1.5.0
panel                                    1.2.3
param                                    1.13.0
parsel                                   1.6.0
parso                                    0.8.3
partd                                    1.4.0
pathlib                                  1.0.1
pathspec                                 0.10.3
patsy                                    0.5.3
pdf2docx                                 0.5.7
pep8                                     1.7.1
pexpect                                  4.8.0
pickleshare                              0.7.5
pillow                                   10.2.0
pip                                      23.2.1
pip-tools                                7.3.0
pkce                                     1.0.3
pkginfo                                  1.9.6
platformdirs                             3.10.0
plotly                                   5.9.0
pluggy                                   1.3.0
ply                                      3.11
posthog                                  3.3.1
poyo                                     0.5.0
prometheus-client                        0.14.1
prompt-toolkit                           3.0.36
Protego                                  0.1.16
protobuf                                 4.23.4
psutil                                   5.9.7
ptyprocess                               0.7.0
pulsar-client                            3.4.0
pure-eval                                0.2.2
py-cpuinfo                               8.0.0
pyaml                                    23.12.0
pyarrow                                  11.0.0
pyarrow-hotfix                           0.6
pyasn1                                   0.4.8
pyasn1-modules                           0.2.8
PyAudio                                  0.2.14
pyautogen                                0.2.4
pycodestyle                              2.10.0
pycosat                                  0.6.4
pycparser                                2.21
pyct                                     0.5.0
pycurl                                   7.45.2
pydantic                                 2.5.2
pydantic_core                            2.14.5
PyDispatcher                             2.0.5
pydocstyle                               6.3.0
pydub                                    0.25.1
pyerfa                                   2.0.0
pyflakes                                 3.0.1
pygame                                   2.5.2
Pygments                                 2.15.0
pygpt-net                                2.0.106
pyinstaller                              5.13.2
pyinstaller-hooks-contrib                2023.12
PyJWT                                    2.4.0
pylint                                   2.16.2
pylint-venv                              2.3.0
pyls-spyder                              0.4.0
PyMuPDF                                  1.23.8
PyMuPDFb                                 1.23.7
pynput                                   1.7.6
pyodbc                                   4.0.34
pyOpenSSL                                23.2.0
pyparsing                                3.0.9
pypdf                                    3.17.4
PyPika                                   0.48.9
pyproject_hooks                          1.0.0
PyQt5-sip                                12.11.0
pyrsistent                               0.18.0
PySide6                                  6.6.1
PySide6-Addons                           6.6.1
PySide6-Essentials                       6.6.1
PySocks                                  1.7.1
pytesseract                              0.3.10
pytest                                   7.4.3
python-dateutil                          2.8.2
python-docx                              0.8.11
python-dotenv                            1.0.0
python-json-logger                       2.0.7
python-lsp-black                         1.2.1
python-lsp-jsonrpc                       1.0.0
python-lsp-server                        1.7.2
python-multipart                         0.0.6
python-slugify                           5.0.2
python-snappy                            0.6.1
python-telegram-bot                      20.3
python-xlib                              0.33
pytoolconfig                             1.2.5
pytorch-triton-rocm                      2.1.0
pytube                                   15.0.0
pytz                                     2023.3.post1
pyviz-comms                              2.3.0
PyWavelets                               1.4.1
pyxdg                                    0.28
PyYAML                                   6.0.1
pyzmq                                    23.2.0
QDarkStyle                               3.0.2
qstylizer                                0.2.2
qt-material                              2.14
QtAwesome                                1.2.2
qtconsole                                5.4.2
QtPy                                     2.2.0
queuelib                                 1.5.0
readme-renderer                          37.3
regex                                    2023.3.23
requests                                 2.31.0
requests-file                            1.5.1
requests-oauthlib                        1.3.1
requests-toolbelt                        0.10.1
responses                                0.13.3
retrying                                 1.3.4
rfc3339-validator                        0.1.4
rfc3986                                  2.0.0
rfc3986-validator                        0.1.1
rich                                     13.3.4
rope                                     1.7.0
rsa                                      4.9
Rtree                                    1.0.1
ruamel.yaml                              0.17.21
ruamel-yaml-conda                        0.17.21
s3fs                                     2023.12.2
safetensors                              0.3.2
scikit-image                             0.20.0
scikit-learn                             1.3.0
scikit-learn-intelex                     20230426.111612
scipy                                    1.11.1
Scrapy                                   2.8.0
seaborn                                  0.12.2
SecretStorage                            3.3.3
semantic-version                         2.10.0
Send2Trash                               1.8.0
service-identity                         18.1.0
setuptools                               68.0.0
shiboken6                                6.6.1
show-in-file-manager                     1.1.4
sip                                      6.6.2
six                                      1.16.0
smart-open                               5.2.1
sniffio                                  1.3.0
snowballstemmer                          2.2.0
sortedcontainers                         2.4.0
soupsieve                                2.4
spark-parser                             1.8.9
SpeechRecognition                        3.10.0
Sphinx                                   5.0.2
sphinxcontrib-applehelp                  1.0.2
sphinxcontrib-devhelp                    1.0.2
sphinxcontrib-htmlhelp                   2.0.0
sphinxcontrib-jsmath                     1.0.1
sphinxcontrib-qthelp                     1.0.3
sphinxcontrib-serializinghtml            1.1.5
spyder                                   5.4.3
spyder-kernels                           2.4.4
SQLAlchemy                               2.0.23
stack-data                               0.2.0
starlette                                0.27.0
statsmodels                              0.14.0
supervision                              0.17.1
sympy                                    1.11.1
tables                                   3.8.0
tabulate                                 0.8.10
TBB                                      0.2
tblib                                    1.7.0
tenacity                                 8.2.3
tensorboard                              2.15.1
tensorboard-data-server                  0.7.2
tensorflow                               2.15.0.post1
tensorflow-estimator                     2.15.0
tensorflow-io-gcs-filesystem             0.34.0
termcolor                                2.4.0
terminado                                0.17.1
text-unidecode                           1.3
textdistance                             4.2.1
threadpoolctl                            2.2.0
three-merge                              0.1.1
tifffile                                 2023.4.12
tiktoken                                 0.5.2
tinycss2                                 1.2.1
tldextract                               3.2.0
tokenizers                               0.15.0
toml                                     0.10.2
tomli                                    2.0.1
tomlkit                                  0.11.1
toolz                                    0.12.0
torch                                    2.1.2+rocm5.6
torchaudio                               2.1.2+rocm5.6
torchvision                              0.16.2+rocm5.6
tornado                                  6.3.2
tqdm                                     4.65.0
traitlets                                5.7.1
transformers                             4.36.2
twine                                    4.0.2
Twisted                                  22.10.0
typer                                    0.9.0
typing_extensions                        4.8.0
typing-inspect                           0.9.0
tzdata                                   2023.4
uc-micro-py                              1.0.1
ujson                                    5.4.0
uncompyle6                               3.7.4
Unidecode                                1.2.0
urllib3                                  1.26.15
uvicorn                                  0.25.0
uvloop                                   0.19.0
w3lib                                    1.21.0
watchdog                                 2.1.6
watchfiles                               0.21.0
wcwidth                                  0.2.5
webencodings                             0.5.1
websocket-client                         0.58.0
websockets                               11.0.3
Werkzeug                                 2.2.3
whatthepatch                             1.0.2
wheel                                    0.38.4
widgetsnbextension                       4.0.5
wikipedia                                1.4.0
wrapt                                    1.16.0
wurlitzer                                3.0.2
xarray                                   2023.6.0
xdis                                     5.0.13
xlrd                                     2.0.1
xxhash                                   2.0.2
xyzservices                              2022.9.0
y-py                                     0.5.9
yapf                                     0.31.0
yarl                                     1.8.2
ypy-websocket                            0.8.2
zict                                     2.2.0
zipp                                     3.15.0
zope.interface                           5.4.0
zstandard                                0.19.0
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
It maybe because of the network proxy software.Pygpt works well on the Windows system with clash.I dont find a useful proxy software on ubuntu.I use qv2ray,but it doesnt work well.Do you have some prxoy softwares to use on ubuntu to recommend?

**Comment by szczyglis-dev** [comment]
I'm afraid I can't recommend a one best proxy for Ubuntu.
The proxy might be the cause, but unfortunately, I can't help in this matter, as the issue of the network connection itself is beyond the scope of PyGPT.
